Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,296 @@
+        self.getStageAndRunBlock {
+            switch self.stage {
+            case .loginFailed:
+                self.delegate?.loginFailed(with: "Incorrect netid and/or password")
+            case .loginScreen:
+                if self.loginCount < 1 {
+                    self.login()
+                }
+            case .finished(let sessionId):
+                self.delegate?.retrievedSessionId(id: sessionId)
+            default: break
+            }
+        }
+    }
+    
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        self.delegate?.loginFailed(with: error.localizedDescription)
+    }
+    
+    /**
+     - Gets the stage enum for the currently displayed web page and runs a block after fetching
+     the HTML for the page.
+     
+     - Does not guarantee Javascript will finish running before the block
+     is executed.
+     */
+    func getStageAndRunBlock(block: @escaping () -> ()) {
+        getHTML(block: { (html: NSString) -> () in
+            if self.failedToLogin() {
+                self.stage = .loginFailed
+            } else if html.contains("<h2>Login OK</h2>") {
+                self.stage = .transition
+            } else if html.contains("<h1>CUWebLogin</h1>") {
+                if self.loginCount < 1 {
+                    self.stage = .loginScreen
+                } else {
+                    self.stage = .loginFailed
+                }
+            } else if self.url?.absoluteString.contains("sessionId") ?? false {
+                guard let url = self.url, let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+                    self.stage = .loginFailed
+                    block()
+                    return
+                }
+
+                if let sessionId = urlComponents.queryItems?.first(where: { $0.name == "sessionId" })?.value {
+                    self.stage = .finished(sessionId: sessionId)
+                } else {
+                    self.stage = .loginFailed
+                }
+            } else {
+                self.stage = .transition
+            }
+            
+            //run block for stage
+            block()
+        })
+    }
+}
+
+class BRBAccountManager {
+    
+    private var connectionHandler: BRBConnectionHandler!
+    var delegate: BRBAccountManagerDelegate?
+    
+    init() {
+        connectionHandler = BRBConnectionHandler()
+        connectionHandler.delegate = self
+    }
+    
+    func saveLoginInfo(loginInfo: LoginInfo) {
+        BRBAccountSettings.saveToKeychain(loginInfo: loginInfo)
+    }
+    
+    func removeSavedLoginInfo() {
+        BRBAccountSettings.removeKeychainLoginInfo()
+    }
+    
+    func getCredentials() -> LoginInfo? {
+        return BRBAccountSettings.loadFromKeychain()
+    }
+    
+    @objc func queryCachedBRBData() {
+        if let (netid, password) = BRBAccountSettings.loadFromKeychain() {
```

Comment:
I think we should rename this: `queryCachedBRBData` sounds like it should notify the delegate with the cached account data. The login data is saved more than it is cached Perhaps `queryBRBDataWithSavedLogin()` might be better.